### ISSUE TITLE
Gitify jumps to another Space

### ISSUE THREAD

**Issue by EricWVGG** [gitify-app/gitify#563]
I am using multiple "Mission Control Spaces." When I hit the menu bar item, instead of the window popping up on the active Space, my Mac jumps to another arbitrarily Space.

This is actually a problem that I'm facing with many apps, ever since "upgrading" to Ventura. Is anyone else here experiencing this issue? I'm slowly going insane, and none of my other popup apps have active enough user communities to determine whether this issue is widespread, or just me and some MacOS plist file…

screen cap: https://user-images.githubusercontent.com/617762/216146490-0b4ebbe5-ee32-4f01-9798-95ae6e3d8b5c.mp4

thanks!

**Comment by JakeSidSmith** [comment]
Hey, @EricWVGG, I'm currently running Monterey and this issue isn't present. I was planning on upgrading in a day or 2, so I'll report back if I have any similar issues. Sounds like it'll likely be an operating system level issue, and so we may not be able to provide a fix. 😢

**Comment by EricWVGG** [comment]
This definitely came up with Ventura. My guess is that if it is an OS bug, it was introduced by that awful new Stage Manager thing… I have dumped loads of Preferences files to reset my Mac UI and gotten nowhere… but yes please if you could use it for a few days in Ventura, I'd love to know.

**Comment by JakeSidSmith** [comment]
Hey, @EricWVGG, finally got round to upgrading.
I'm now on MacOS 13.2.1 and it doesn't skip away from full screen apps when I click Gitify. Are you on the same MacOS version as me? If not, maybe try an upgrade.

**Comment by tomchkk** [comment]
I've also had this issue since updating. Currently on MacOS 13.2.1 and it's still present for me. Restarting Gitify fixes it, but only for so long...

**Comment by EricWVGG** [comment]
Thanks for chiming in, @tomchkk . You're the first other person I've encountered with this problem.

I'm on MacOS 13.3. Still present. I recently tried dumping _all_ of my com.apple.* entries in `/Library/Preferences`, still no dice. I'm going to try running from a brand new user account when work slows down a little (next month, I hope).

**Comment by EricWVGG** [comment]
@JakeSidSmith Just to clarify, it doesn't just jump away from full-screen apps, it also jumps away from various Spaces. It's like my popup apps have picked one space as their "home" and always want to go there.

**Comment by JakeSidSmith** [comment]
I've just had this take me to my main desktop for the first time. 🤔 Restarting it has (at least temporarily) fixed it.

That makes me think that this is something that _could_ be fixed within Gitify, but it could still be a lower level electron issue, or something to do with Ventura.

**Comment by JakeSidSmith** [comment]
Having a look back through the history, it appears that a crucial line of code was removed during some changes to allow people to switch between having a popover window, and a regular windowed app.

https://github.com/manosim/gitify/commit/3aadb382c709aea68f7ef849f038fc5b35709f96#diff-58417e0f781b6656949d37258c8b9052ed266e2eb7a5163cad7b0863e6b2916aL146

And it was never replaced when that functionality was removed...

So it may have been purely by chance that it has worked as we all expected up til now. 😂 Unless the new implementation _should_ just open on the current window anyway. 🤷

**Comment by EricWVGG** [comment]
"when that functionality was removed" could you elaborate on that?

I'm trying to figure out if this is a Ventura change. Gitify is actually the least of my problems, I have popup apps that I use dozens of times per hour. I'm losing my mind on this.

**Comment by JakeSidSmith** [comment]
The app became a regular old windowed app in v2 (https://github.com/manosim/gitify/releases/tag/2.0.0).
And was moved back to a tray icon in v3 (https://github.com/manosim/gitify/releases/tag/v3.0.0).

I believe the crucial line of code was removed in v2, and never replaced in v3.

**Comment by JakeSidSmith** [comment]
That said, I've just tested out a few of my other tray icon apps...

Aside from Gitify, I have another 2 apps that work in a similar way. 1 of these directed me away from my full screen apps (which I don't remember it doing previously), while the other one still correctly overlays.
After restarting my Mac, none of them directed me away from my full screen apps.

Although I think we could add a line of code to Gitify that _may_ fix the problem, it seems that there is some kind of underlying issues with MacOS, so that won't help you with all of the other apps taking you away constantly.

**Comment by EricWVGG** [comment]
yeah this problem always goes away after a fresh reboot, but only for an hour or so

one of these days I'm going to take the nuclear option and just delete my user account and start fresh (I like to do that about once every five years anyway), but not until work slows down. le sigh

Thanks for looking into this for me.

**Comment by stackptr** [comment]
I've also been encountering this issue, where Gitify seems to "stick" to a certain space. I don't think this has happened during a session, i.e., as I'm actively using the device, but seems to happen when I log back in to macOS. At least a few times a week, I have to restart Gitify until it functions as expected.

Experiencing this issue just now prompted me to check GitHub and also experiment with the app. I use dual monitors, with each monitor using a different set of spaces, and typically only interact with Gitify on one of them. However, I find that I can reproduce a type of "stickiness" if I do the following:

- Click Gitify menubar icon on one monitor to open the window
- Click anywhere on the same monitor to close the Gitify window, but _not on the menubar icon_
- Immediately click the Gitify menubar icon on the other monitor
- Window opens on the previous monitor, rather than the monitor of my cursor

If the first window is closed by clicking the menubar icon, or the other monitor is interacted with prior to trying to instantiate the window, then the window opens in the right monitor.

If the above is the same issue as the window opening in a different space in the same monitor, then this behavior might be an issue in macOS; I can reproduce this with the battery icon as well as with SoundSource.

**Comment by afonsojramos** [comment]
I suspect that the issue here is with the `menubar` config, namely the `preloadWindow` and the `delayedHideAppIcon` function call. Will need some further looking into for sure though.

**Comment by adufr** [comment]
I am not experiencing this bug anymore! (🎉)

I think that the rollback to electron@19 might have helped?

**Comment by setchy** [comment]
@EricWVGG - I'd love to hear if this issue still exists on gitify@5.0.0

**Comment by bmulholland** [comment]
I still get it on v5

**Comment by adufr** [comment]
@graemer957 Agreed, looks like this time the electron update has fixed this issue for good.
I think it can be closed as completed.

**Comment by bmulholland** [comment]
Cool, then closing this for now. We can re-open if there's a new report.

**Comment by setchy** [comment]
Fantastic news 🥳

**Comment by bmulholland** [comment]
This is happening for me again :(
v5.2.0

**Comment by tomchkk** [comment]
Happened for me again once today as well 😢 
v.5.2.0

**Comment by bmulholland** [comment]
Yep, restarting fixes it. I suspect sleep or something triggers the problem.

**Comment by JakeSidSmith** [comment]
> Yep, restarting fixes it. I suspect sleep or something triggers the problem.

Just make sure you've had plenty of caffeine. 😜

**Comment by tomchkk** [comment]
It's back! 😨 Happened yet again - yes restarting fixes it, but only until the next time... 😭

**Comment by schester44** [comment]
+1 here, this happens for me from time to time. currently on 5.8.1

**Comment by setchy** [comment]
Thanks for all the confirmed sightings.  

We'll open an upstream GitHub issue with max-mapper/menubar shortly shortly and x-ref here.

**Comment by JakobEdding** [comment]
I found a workaround which allowed me to avoid this issue for 2 weeks and counting:
1. With Gitify running, drag & drop /Applications/Gitify.app into the Dock
2. Right-click Gitify in the Dock
3. Options > Assign to: All Desktops
4. (Remove Gitify from the Dock again)

**Comment by setchy** [comment]
Interestingly, looks like menubar already sets this via their default `showOnAllWorkspaces` [option](https://github.com/max-mapper/menubar/blob/6aa6df7aa8f6e5758aa5e48a3020caad264bfa73/src/Menubar.ts#L300)


https://github.com/max-mapper/menubar/blob/6aa6df7aa8f6e5758aa5e48a3020caad264bfa73/src/Menubar.ts#L300-L305

**Comment by akellbl4** [comment]
A little more info on the issue.
When I'm in a space where I use fullscreen mode Gitify will move me to a space without full screen mode, but it works fine on all non-fullscreen spaces without the workaround from @JakobEdding. The issue persists even if the workaround is applied.

macOS: 15.3
Gitify: 6.1.0

**Comment by Notaduck** [comment]
> A little more info on the issue. When I'm in a space where I use fullscreen mode Gitify will move me to a space without full screen mode, but it works fine on all non-fullscreen spaces without the workaround from @JakobEdding. The issue persists even if the workaround is applied.
> 
> macOS: 15.3 Gitify: 6.1.0
> 
> UPD: sometimes the dropdown will close itself after moving to another space.

I've the exact same issue as you with the same version of macOS and Gitigy.

**Comment by samsternatmiter** [comment]
+1 to this issue

* macOS 26.0.1 (25A362)
* Gitify v6.20.0

**Comment by afonsojramos** [comment]
I believe that this is no longer an issue on v7. Can someone else validate?

**Comment by basurahombre** [comment]
Still happens on v7.4.0

Screen recording showing switching from Firefox space to Ghostty space:

https://github.com/user-attachments/assets/060a8b0c-872a-40c2-9f47-00f675a7a96c

It seems to get stuck over time. On first start, Gitify will show up on any space. Then it tends to get stuck to a specific space at some point. I haven't noticed a specific pattern yet.

**Comment by tmcknight** [comment]
Been digging into this on macOS 27.0 (arm64), 7.8.0. The popup gets bound to one Space and drags you there.

Repro:
1. Put an app fullscreen (I used Ghostty).
2. Click the tray icon from that fullscreen Space so the popup opens there.
3. Quit the fullscreen app.
4. Go to another fullscreen app's Space.
5. Click the tray icon → yanked to the desktop. Keeps happening from then on.

This seems to reproduce the bug reliably for me. And might explain why restarting Gitify fixes it temporarily (new window, fresh Space assignment) and why it comes back after sleep, login, or fullscreen churn.

I did some digging with Claude, and I don't see a fix from inside Gitify. Looks like it belongs in `electron-menubar`, or is an Electron/macOS limitation.

**Comment by afonsojramos** [comment]
@tmcknight `electron-menubar` is certainly where the bug needs to be fixed, but `electron-menubar` is also a fork of `menubar`, which also always had this issue. We've had attempts at fixing this, but without much luck - since you're still seeing it. 
If you encounter this often, I would ask you to maybe run a local build with local electron-menubar and try reproducing the issue and, if possible, fix it, because I can't seem to face this issue as often as others I reckon.
